File: public/examples/integrations/toolkits/search/google_flights/search_roundtrip_flights_example_llm_oai.py
import os
from openai import OpenAI

USER_ID = "you@example.com"
PROMPT = "Search for roundtrip flights from Los Angeles to San Francisco on September 1st, 2025, returning on September 5th, 2025."
TOOL_NAME = "Search.SearchRoundtripFlights"

client = OpenAI(
    base_url="https://api.arcade.dev", api_key=os.environ.get("ARCADE_API_KEY")
)

response = client.chat.completions.create(
    messages=[
        {"role": "user", "content": PROMPT},
    ],
    model="gpt-4o-mini",
    user=USER_ID,
    tools=[TOOL_NAME],
    tool_choice="generate",
)
print(response.choices[0].message.content)

File: public/examples/integrations/toolkits/search/google_flights/search_roundtrip_flights_example_llm_oai.js
import OpenAI from "openai";

const USER_ID = "you@example.com";
const PROMPT =
	"Search for roundtrip flights from Los Angeles to San Francisco on September 1st, 2025, returning on September 5th, 2025.";
const TOOL_NAME = "Search.SearchRoundtripFlights";

const client = new OpenAI({
	baseURL: "https://api.arcade.dev",
	apiKey: process.env.ARCADE_API_KEY,
});

const response = await client.chat.completions.create({
	messages: [{ role: "user", content: PROMPT }],
	model: "gpt-4o-mini",
	user: USER_ID,
	tools: [TOOL_NAME],
	tool_choice: "generate",
});

console.log(response.choices[0].message.content);
